test/bulk_insert_test.rb

```diff
@@ -34,5 +34,18 @@ class BulkInsertTest < ActiveSupport::TestCase
 
     assert_equal Testing.default_bulk_columns, default_columns
   end
+  
+  test "raises if on_duplicate_key is not a hash" do
+    assert_raise { Testing.bulk_insert(on_duplicate_key: 2) }
+  end
+
+  test "raises if on_duplicate_key is a hash without update" do
+    assert_raise { Testing.bulk_insert(on_duplicate_key: { delete: 2 }) }
+  end
 
+  test "accepts on_duplicate_key argument and constructs sql_on_duplicate in the worker" do
+    worker = Testing.bulk_insert(on_duplicate_key: { update: '123'})
+
+    assert_equal worker.instance_eval { @sql_on_duplicate }, ' on duplicate key update 123'
```

Comment:
The more I think about this, the more I feel uncomfortable about peeking at an object's internal state. Rather than reaching inside the object and violating encapsulation, I'd like to see another way to test this. I know SQLite doesn't support the feature, but maybe there's another way to do this. PR #21 is proposing a refactoring that would expose a method that returns the generated SQL. Maybe this test could query that SQL to make sure that the expected snippet exists? You'll need to wait until that PR lands, but I think it would expose a better way to implement this test.